### ISSUE TITLE
Random Boxes Are Open-world Object Detectors

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#46]
### Paper Link
https://openaccess.thecvf.com/content/ICCV2023/papers/Wang_Random_Boxes_Are_Open-world_Object_Detectors_ICCV_2023_paper.pdf

### Code Link
https://github.com/scuwyh2000/RandBox

## Motivation

**Known-FG (foreground)**: calculate the matching scores between the predicted region proposal labels and the groundtruth labels of known objects, where the label is a pair of class and bbox (bounding box. For the matched proposals, the loss is class cross-entropy $L_{cls}$ and bbox regression $L_{reg}$ .

**BG (background)**: all the mismatched proposals are considered background, whose loss is only $L_{cls}$ with the label ”BG”;

**Unknown-FG**: for the mismatched proposals with class confidence larger than a threshold probably unknown objects—the loss is only $L_{cls}$ with a special label “unknown” w/o bbox regression.

**Key Problem**: the mismatched proposals, mixed with both unknown objects and BG, all have low confidence, losing a gradual confidence decrease to differentiate Unknown-FG from BG (the red and black colors are inseparable).

<img width="600" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/20a637c5-5d43-49d3-8078-9fd342f1d1ca">

**Why? Confounding Effect**: the training data with limited known object labels that mislead the feature transfer.

**Solution**: inviting an **independent 3rd-party proposal sampler** to explore more possible locations of unknown objects.

<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/9d58b3b3-d549-47b6-bf94-d8a8e594fcd1">

## Methodology

<img width="1349" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/c0e6ce8b-90fa-4079-9525-43c01bc77a09">

### Detector with Random Proposals.
**Training**
- Randomly generate 500 bboxes as region proposals
- For each bbox, we sample 4 random real numbers in \[0, 1\], drawn from the standard Gaussian distribution, correspinding to coordinates and size (proportional to image size). The width and height of out-of-boundary bboxes are adjusted to stay in the image.

**Testing**
- use 10,000 pre-defined bboxes as proposals on each image, which cover 10 scales, 10 aspect ratios and 100 spatial locations
- use NMS

### Dynamic-k m Matcher for Known-FG
Each ground-truth bbox b is matched with k proposals based on the bipartite matching score, where k is dynamically selected as the sum of the IoU between each proposal and b.

### Matching Score for Unknown-FG
- Top N proposals with the largest matching score. Among the selected proposals, those already in Known-FG are removed to form our Unknown-FG subset.

<img width="200" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/6c7628e3-672b-4c25-b670-6fcec5286d34">

- the top-N (largest matching score) proposals that are not in KnownFG are pseudo-labeled as “unknown” to train the UnknownFG logit.
- As shown in Figure 4b, an unknown object proposal can have non-trivial likelihood of belonging to a known class due to shared feature (dashed arrow), while BG proposal almost shares no feature with known.
- Our matching score fairly evaluates each proposal by checking both known classes (first | K | logits), as well as “unknown” ( | K | +1-th logit). Hence unknown object proposals can be reliably selected in the Unknown-FG subset with high likelihood of belonging to “unknown”.

### Theory?

<img width="600" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/d5a49451-25c3-4808-9c52-cf161eb21818">

<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/551de615-5653-4e22-8cfd-8c2961fea4e6">

## Experiments 

<img width="1384" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/07012139-3617-4ab9-b43c-7ab563f216d5">
<img width="1429" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/47cf2743-dfb2-4a6b-b9b9-54955778188e">
<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/4b3d596e-8eec-4bcd-b83c-8ead6f98ee98">
<img width="1130" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/6c7b1b6e-c68e-4aa3-bea2-12f72bdf7ec8">
<img width="1221" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/3c3c7db8-2296-49af-b4c4-99eb65428311">
